### PR TITLE
New package: ForTrilinos

### PR DESCRIPTION
**Pull request by sethrj**
Successfully installed and ran tests for ForTrilinos on mac:
```
-- darwin-catalina-x86_64 / apple-clang@11.0.0 ------------------
thsdvhq fortrilinos@2.0.0-dev2+hl+shared build_type=RelWithDebInfo
whhv6v4     trilinos@12.18.1~adios2~alloptpkgs+amesos+amesos2+anasazi+aztec+belos~boost~cgns~chaco~complex~debug~dtk+epetra+epetraext~exodus+explicit_template_instantiation~float+fortran~glm~gtest~hdf5~hypre+ifpack+ifpack2~intrepid~intrepid2~isorropia+kokkos~matio~mesquite~metis~minitensor+ml+mpi+muelu~mumps~netcdf+nox~openmp~phalanx~piro~pnetcdf~python~rol~rythmos+sacado~shards+shared~shylu~stk+stratimikos~suite-sparse~superlu~superlu-dist~teko~tempus+teuchos+tpetra~x11~xsdkflags~zlib+zoltan+zoltan2 build_type=RelWithDebInfo gotype=long_long
gkdwhwe         openblas@0.3.10~consistent_fpcsr~ilp64+pic+shared threads=none
rq44ux2         openmpi@3.1.6~atomics~cuda~cxx~cxx_exceptions+gpfs~java~legacylaunchers~memchecker~pmi~singularity~sqlite3+static~thread_multiple+vt+wrapper-rpath fabrics=none schedulers=none
d3sdlmx             hwloc@1.11.11~cairo~cuda~gl~libudev+libxml2~netloc~nvml~pci+shared
zd36xsu                 libxml2@2.9.10~python
dacit3k                     libiconv@1.16
27pku55                     xz@5.2.5
fwg66ug                     zlib@1.2.11+optimize+pic+shared
```
and successfully tested a downstream app against the installed ForTrilinos.

@keitat This removes the built-in `fortrilinos` variant (and resource) since ForTrilinos is now a separate library downstream of Trilinos.

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -250,6 +243,7 @@ class Trilinos(CMakePackage):
     conflicts('+belos', when='~teuchos')
     conflicts('+epetraext', when='~epetra')
     conflicts('+epetraext', when='~teuchos')
+    conflicts('+exodus', when='~netcdf')
```
Comment (aprokop) [inline var/spack/repos/builtin/packages/trilinos/package.py:246]: Not really part of this PR.

**Review by aprokop — APPROVED** [review]

File: var/spack/repos/builtin/packages/trilinos/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/trilinos/package.py:246]: Separate commit, though? Could rebase and merge. Found while testing fortrilinos.

Comment (aprokop) [reply var/spack/repos/builtin/packages/trilinos/package.py:246]: This is fine.
